app/Http/Controllers/DashboardController.php

```diff
@@ -29,36 +27,42 @@ public function __construct()
      */
     public function index()
     {
-        try
-        {
-            //Todo use the users preferred device
-            //Get the first device that doesn't have a null location_id
-            $device = Device::publicDashData()->where('location_id', '!=', 'null')->orderBy('id', 'ASC')->first();
-        }
-        catch(ModelNotFoundException $e)
-        {
-            //If no device is found then redirect the user to the home page and display a message about registering a device
-            return redirect('home')->with('no_devices', 'To access the dashboard page there must be at least one registered device assigned to a location.');
-        }
-        
-        //Get the devices location
-        $location = $device->location()->select('id', 'name', 'site_id')->firstOrFail();
-        //Get the devices site
-        $site = $location->site()->select('id', 'name')->firstOrFail();
+        //TODO use the users preferred device
+        $device_id = Device::firstOrFail()->id;
     
-        $data = $this->dashData($site, $location, $device);
-        
-        return view('dashboard.index', [ 'active_device' => $data[0], 'devices' => $data[1], 'locations' => $data[2], 'sites' => $data[3] ]);
-    }
+        $location = Location::where('id', '=', $device_id)->first();
+        $site_id = $location->site_id ?? 0;
+        $location_id = $location->id ?? 0;
     
-    /**
-     * Show the development layouts for dashboard.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function dev_layout()
-    {
-        return view('dashboard.dev_layout');
+        //Get all sites with the selected site first
+        $sites = Site::select('id', 'name')
+            ->orderByRaw("id = ? DESC", $site_id)
+            ->orderBy('name', 'ASC')
+            ->get();
```

Comment:
This can just be $device->site->where(id, $device->location->site->id)->orderBy('name', 'ASC')->get();

Reply:
I had a mistake earlier with using `Device::firstOrFail()` when I should have been using `Device::first()`(eventually `Device::find()` when using a user's preferred device). Since there might be the possibility that there is no device registered yet. If I do what you said I will receive an error since $device will be null. Also I need to use `->orderByRaw("id = ? DESC", $site_id)` to have the site matching `$site_id` first in the collection.